apps/snitch_core/lib/core/data/model/order.ex

```diff
@@ -216,4 +216,28 @@ defmodule Snitch.Data.Model.Order do
   defp update_line_item_costs(line_items) when is_list(line_items) do
     LineItemModel.update_unit_price(line_items)
   end
+
+  def get_order_count_by_state(start_date, end_date) do
+    Order
+    |> where([o], o.inserted_at >= ^start_date and o.inserted_at <= ^end_date)
+    |> group_by([o], o.state)
+    |> order_by([o], asc: o.state)
+    |> select([o], %{state: o.state, count: count(o.id)})
+    |> Repo.all()
+  end
+
+  defmacro to_char(field, format) do
+    quote do
+      fragment("to_char(?, ?)", unquote(field), unquote(format))
+    end
+  end
```

Comment:
@anantanant2015 this is code repetition, can you put this in the tools dir and use the common logic everywhere.

Reply:
@ashish173 I have tried putting this logic in helper for query, and after that compilation errors are coming as this is not valid expression.